specification.markdown

```diff
@@ -155,11 +155,39 @@ Class D    | 4      | 0x101FFFFF - 0x1FFFFFFE |   268,338,304
+## Unknown Data Size
+
+An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)).
+
+The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first:
+
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a Parent Element of the Unknown-Sized Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is not a Descendant Element of the Unknown-Sized Element but share a common direct parent. For example a Top-Level Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a root element.
+- The end of the EBML Document, either when reaching the end of the file or because a new EBML Header started.
+
+Consider an Unknown-Size Element which EBML path is `\root\level1\level2\elt`. Here are some possible and impossible way to end this element:
+
+EBML Path of next element          | Status
+-----------------------------------|------------------------------
+`\root\level1\level2`              | Ends the unknown-size element, as it is a new Parent element
+`\root\level1`                     | Ends the unknown-size element, as it is a new Parent element
+`\root`                            | Ends the unknown-size element, as it is a new root element
+`\root2`                           | Ends the unknown-size element, as it is a new root element
```

Comment:
So is 'root2' supposed to be a new Root Element, since as in two concatenated EBML Documents? Perhaps an xpath style of `root[2]` would be more intuitive than root2.

Reply:
All elements in the table are the `EBML Path of next element`. `\root` has the same ID as the root in `\root\level1\level2\elt` but it's a new element.

`\root2` is a different ID than the root in `\root\level1\level2\elt`.